PWGJE/TableProducer/emcalClusterHadronicCorrectionTask.cxx

```diff
@@ -0,0 +1,309 @@
+
+      int Nmatches = 0;         // counter for closest matched track
+      double closestTrkP = 0.0; // closest track momentum
+      double totalTrkP = 0.0;   // total track momentum
+
+      // pT-dependent track-matching instead of PID based track-matching to be adapted from Run 2 - suggested by Markus Fasel
+
+      TF1 funcPtDepEta("func", "[1] + 1 / pow(x + pow(1 / ([0] - [1]), 1 / [2]), [2])");
+      funcPtDepEta.SetParameters(Eta0, Eta1, Eta2);
+      TF1 funcPtDepPhi("func", "[1] + 1 / pow(x + pow(1 / ([0] - [1]), 1 / [2]), [2])");
+      funcPtDepEta.SetParameters(Phi0, Phi1, Phi2);
+
+      // No matched tracks (trackless case)
+      if (tracksofcluster.size() == 0) {
+        // Use original cluster energy values, no subtraction needed.
+        registry.fill(HIST("h2_ClsEvsNmatches"), cluster.energy(), 0);
+        registry.fill(HIST("h_Ecluster1"), Ecluster1);
+        registry.fill(HIST("h_Ecluster2"), Ecluster2);
+        registry.fill(HIST("h_EclusterAll1"), EclusterAll1);
+        registry.fill(HIST("h_EclusterAll2"), EclusterAll2);
+        registry.fill(HIST("h2_ClsEvsEcluster1"), cluster.energy(), Ecluster1);
+        registry.fill(HIST("h2_ClsEvsEcluster2"), cluster.energy(), Ecluster2);
+        registry.fill(HIST("h2_ClsEvsEclusterAll1"), cluster.energy(), EclusterAll1);
+        registry.fill(HIST("h2_ClsEvsEclusterAll2"), cluster.energy(), EclusterAll2);
+        hadroniccorrectedclusters(Ecluster1, Ecluster2, EclusterAll1, EclusterAll2);
+        continue;
+      }
+
+      // Looping over all matched tracks for the cluster
+      // Total number of matched tracks = 20 (hard-coded)
+      for (const auto& match : tracksofcluster) {
+
+        double mom = abs(match.track_as<myTracks>().p());
+        registry.fill(HIST("h_matchedtracks"), 1);
+
+        // CASE 1: skip tracks with a very low pT
+        if (mom < 1e-6) {
+          continue;
+        } // end CASE 1
+
+        // CASE 2:
+        //  a) If one matched track -> 100% energy subtraction
+        //  b) If more than one matched track -> 100% energy subtraction
+        //  c) If you want to do systematic studies -> perform the above two checks a) and b), and then subtract 70% energy instead of 100%
+
+        // Perform dEta/dPhi matching
+        double dEta = match.track_as<myTracks>().eta() - cluster.eta();
+        double dPhi = TVector2::Phi_mpi_pi(match.track_as<myTracks>().phi() - cluster.phi());
+
+        // Apply the eta and phi matching thresholds
+        // dEta and dPhi cut : ensures that the matched track is within the desired eta/phi window
+
+        // Do pT-dependent track matching
+        if (doMomDepMatching) {
+          auto trackEtaMax = funcPtDepEta.Eval(mom);
+          auto trackPhiHigh = +funcPtDepPhi.Eval(mom);
+          auto trackPhiLow = -funcPtDepPhi.Eval(mom);
+
+          if ((dPhi < trackPhiHigh && dPhi > trackPhiLow) && fabs(dEta) < trackEtaMax) {
+            if (Nmatches == 0) {
```

Comment:
while the list is indeed sorted to contain the closest track first, for safety reasons I would explicitly check that it is the closest of the ones you looped over